### PR TITLE
UTY-1176: Code Generation Workflow Tweaks

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Added a marker file to `Temp` to indicate auto codegen has run already. This should stop the constant popups every recompile.

This gets removed when Unity closes so it will always generate when first started.
#### Tests
Tested workflow locally - worked nicely.
#### Documentation
N/A 
#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.
@jared-improbable

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.tools/GenerateCode.cs [hunk 1]
```diff
@@ -20,20 +20,33 @@ internal static class GenerateCode
         private static readonly string SchemaCompilerRelativePath =
             $"../build/CoreSdk/{Common.CoreSdkVersion}/schema_compiler/schema_compiler";
 
+        private static readonly string StartupCodegenMarkerFile =
+            Path.GetFullPath("Temp/ImprobableCodegen.marker");
```
Comment (firtina-improbable) [inline workers/unity/Packages/com.improbable.gdk.tools/GenerateCode.cs]: `/`, may be better to use `Path.Join`

Comment (firtina-improbable) [reply workers/unity/Packages/com.improbable.gdk.tools/GenerateCode.cs]: I think it's safe to use `"Temp"` here but may be even safer to do e.g. `Path.Join(Application.dataPath, "..", "Temp", "filename.ext")`

Comment (callumlawson) [reply workers/unity/Packages/com.improbable.gdk.tools/GenerateCode.cs]: if this is the right folder, we should use this:
https://docs.unity3d.com/ScriptReference/Application-temporaryCachePath.html

Comment (firtina-improbable) [reply workers/unity/Packages/com.improbable.gdk.tools/GenerateCode.cs]: This one could potentially clash with other projects:

`%USERPROFILE%/AppData/Local/Temp/{company name}/{project name}`

Where {project name} is likely `unity` and {company name} is likely `DefaultCompany`

Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.tools/GenerateCode.cs]: That shouldn't be the windows temp, but the project temp

Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.tools/GenerateCode.cs]: There's an easy way to test this 😉

Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.tools/GenerateCode.cs]: It is indeed - `%USERPROFILE%/AppData/Local/Temp/{company name}/{project name}`

Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.tools/GenerateCode.cs]: It also doesn't get cleaned when Unity closes or opens. Am going to revert back to the original

Comment (firtina-improbable) [reply workers/unity/Packages/com.improbable.gdk.tools/GenerateCode.cs]: re: the original: reminder about my comments about Path.Join

**Pull request merged**